discovery-installer.spec

```diff
@@ -1,48 +1,71 @@
 Release:        1%{?dist}
 Epoch:          0
 
 License:        GPLv3
-URL:            https://github.com/quipucords/discovery-quadlets
+URL:            https://github.com/quipucords/discovery-installer
 Source0:        %{url}/archive/%{version}.tar.gz
 
 BuildArch:      noarch
 
 Requires:       bash
 
 %description
-discovery-installer configures and installs the quipucords server to be
+%{name} configures and installs the %{product_name_title} server to be
 managed and run via systemd using Podman Quadlet services.
 
 %prep
-%autosetup -n discovery-quadlets-%{version}
+# Note: this must match the GitHub repo name. Do not substitute variables.
+%autosetup -n discovery-installer-%{version}
 
 %install
 mkdir -p %{buildroot}/%{_bindir}
 mkdir -p %{buildroot}/%{_datadir}/%{name}/bin
-cp bin/%{name} %{buildroot}/%{_bindir}/%{name}
+cp bin/discovery-installer %{buildroot}/%{_bindir}/%{name}
 cp bin/create-server-password %{buildroot}/%{_datadir}/%{name}/bin/
 cp bin/create-app-secret %{buildroot}/%{_datadir}/%{name}/bin/
-mkdir -p %{buildroot}/%{_datadir}/%{name}/config
-cp config/*.container config/*.network %{buildroot}/%{_datadir}/%{name}/config/
+
 mkdir -p %{buildroot}/%{_datadir}/%{name}/env
 cp env/*.env %{buildroot}/%{_datadir}/%{name}/env/
 
+# Copy and rename original source files with appropriate branding.
+mkdir -p %{buildroot}/%{_datadir}/%{name}/config
+cp config/discovery-app.container %{buildroot}/%{_datadir}/%{name}/config/%{product_name_lower}-app.container
+cp config/discovery-celery-worker.container %{buildroot}/%{_datadir}/%{name}/config/%{product_name_lower}-celery-worker.container
+cp config/discovery-db.container %{buildroot}/%{_datadir}/%{name}/config/%{product_name_lower}-db.container
+cp config/discovery-redis.container %{buildroot}/%{_datadir}/%{name}/config/%{product_name_lower}-redis.container
+cp config/discovery-server.container %{buildroot}/%{_datadir}/%{name}/config/%{product_name_lower}-server.container
+cp config/discovery.network %{buildroot}/%{_datadir}/%{name}/config/%{product_name_lower}.network
+
+# Update source files contents with appropriate branding.
+sed -i 's/Discovery/%{product_name_title}/g;s/discovery/%{product_name_lower}/g' %{buildroot}/%{_bindir}/%{name}
+sed -i 's/Discovery/%{product_name_title}/g;s/discovery/%{product_name_lower}/g' %{buildroot}/%{_datadir}/%{name}/bin/*
+sed -i 's/Discovery/%{product_name_title}/g;s/discovery/%{product_name_lower}/g' %{buildroot}/%{_datadir}/%{name}/config/%{product_name_lower}*
+sed -i 's/Discovery/%{product_name_title}/g;s/discovery/%{product_name_lower}/g' %{buildroot}/%{_datadir}/%{name}/env/*
+
+# Inject specific image versions into the container files.
+sed -i 's/^Image=.*/Image=quay.io\/quipucords\/quipucords:1.8/g' %{buildroot}/%{_datadir}/%{name}/config/%{product_name_lower}-server.container
+sed -i 's/^Image=.*/Image=quay.io\/quipucords\/quipucords:1.8/g' %{buildroot}/%{_datadir}/%{name}/config/%{product_name_lower}-celery-worker.container
```

Comment:
luv it!

can't we use `%{version}` tag here instead of hardcoding to 1.8?

Reply:
We can't use `%{version}` if we want the images to float on `X.Y` instead of pinning to `X.Y.Z`, but I can define a _new_ variable that's just `X.Y`. I had considered doing that before, but I wanted to wait until we had more discussion tomorrow. Since you had the same idea, no need to wait!